src/compiler/type_check.cc

```diff
@@ -103,12 +107,61 @@ class TypeChecker : public ReturningVisitor<Type> {
     if (node->has_body()) visit(node->body());
     return Type::invalid();
   }
+  Type visit_Global(Global* node) {
```

Comment:
All these methods always return Type::invalid(), so they are only after the side effects. It seems a little bit surprising at first glance, so maybe it makes sense to think of a good place to document that?

Reply:
added a comment.
done.